aten/src/ATen/native/cuda/MixedDtypesLinear.cu

```diff
@@ -0,0 +1,354 @@
+#include <ATen/ATen.h>
+#include <ATen/core/Tensor.h>
+#include <ATen/cuda/CUDAUtils.h>
+
+#if defined(USE_ROCM) || defined(_MSC_VER) || (defined(CUDA_VERSION) && CUDA_VERSION < 11800)
+// Doesn't work on ROCm or Windows yet
+// TODO: Add compiler warning? Add PyTorch config flag?
+#else
+#include <cuda_runtime.h>
+#include <cutlass/cutlass.h>
+#include <cutlass/tensor_ref.h>
+
+#include <cutlass/gemm/device/gemm_universal_base.h>
+#include <cutlass/gemm/kernel/default_gemm.h>
+
+#include <ATen/native/cuda/cutlass_extensions/epilogue_helpers.h>
+#include <ATen/native/cuda/cutlass_extensions/gemm/kernel/default_fpA_intB_traits.h>
+#include <ATen/native/cuda/cutlass_extensions/gemm/kernel/fpA_intB_gemm.h>
+#include <ATen/native/cuda/cutlass_extensions/gemm/threadblock/default_mma.h>
+#endif
+
+#if defined(USE_ROCM) || defined(_MSC_VER) || (defined(CUDA_VERSION) && CUDA_VERSION < 11800)
+// Doesn't work on ROCm or Windows yet
+#else
+#define CUTLASS_STATUS_CHECK(status)                                      \
+  {                                                                       \
+    TORCH_CHECK(status == cutlass::Status::kSuccess,                      \
+                "Got CUTLASS error: ", cutlassGetStatusString(status));   \
+  }
+#endif
+
+namespace at {
+namespace native {
+
+#if defined(USE_ROCM) || defined(_MSC_VER) || (defined(CUDA_VERSION) && CUDA_VERSION < 11800)
+// Doesn't work on ROCm or Windows yet or old compiler
+#else
+template<typename ElementInputA, typename ElementInputB, typename EpilogueTag>
+Tensor
+mixed_dtypes_linear_cutlass(
+    const Tensor& input, const Tensor& weight, const Tensor& scale,
+    const Tensor& bias) {
+  const int length_m = input.size(0);
+  const int length_k = weight.size(0);
+  const int length_n = scale.size(0);
+
+  using ElementOutput = ElementInputA;
+
+  using SmArch = cutlass::arch::Sm80;
+  using ThreadblockShape = cutlass::gemm::GemmShape<32, 128, 64>;
+  using WarpShape = cutlass::gemm::GemmShape<32, 32, 64>;
+  using InstructionShape = cutlass::gemm::GemmShape<16, 8, 16>;
+  using ThreadblockSwizzle = cutlass::gemm::threadblock::GemmIdentityThreadblockSwizzle<>;
+  using Operator = cutlass::arch::OpMultiplyAddDequantizeInterleavedBToA;
+
+  constexpr auto ThreadblockK = 64;
+  constexpr auto ElementsPerCacheLine = 128 * 8 / cutlass::sizeof_bits<ElementInputB>::value;
+  constexpr auto ColumnsInterleaved   = ElementsPerCacheLine / ThreadblockK;
+
+  using LayoutInputA = cutlass::layout::RowMajor;
+  using LayoutInputB = cutlass::layout::ColumnMajorTileInterleave<ThreadblockK, ColumnsInterleaved>;
+  using LayoutOutput = LayoutInputA;
+
+  constexpr auto ElementsPerAccessA = 128 / cutlass::sizeof_bits<ElementInputA>::value;
+  constexpr auto ElementsPerAccessB = 128 / cutlass::sizeof_bits<ElementInputB>::value;
+  constexpr auto ElementsPerAccessC = ElementsPerAccessA;
+  constexpr auto Stages = 4;
+  constexpr auto SplitKFactor = 1; // Wrong outputs if !=1, even if
```

Comment:
Hi, I am encounting the same question,SplitKFactor must be 1, otherwise, I'll get a error. Do you know why? I am looking forward to your reply so much, Thank you!